tap_pardot/sync.py

```diff
@@ -16,3 +18,37 @@ def sync(client, config, state):
 
         for rec in stream_object.sync():
             singer.write_record(stream_id, rec)
+
+
+def sync_properties(client: Client):
+    streams = [
+        {
+            "stream": "prospectAccountFields",
+            "path": "prospectAccount/version/4/do/describe",
+            "data_path": ["result", "field"],
+        },
+        {
+            "stream": "customFields",
+            "path": "customField/version/4/do/query",
+            "data_path": ["result", "customField"],
+        },
+    ]
+    for stream in streams:
+        path = stream["path"]
+        data = client._make_request(
+            "GET",
+            f"{ENDPOINT_BASE}{path}",
+            params={"format": "json"},
+        )
+        records = get_data(data, stream["data_path"])
+        singer.write_records(stream["stream"], records)
+
+
+def get_data(data: Dict, path: List) -> Dict:
+    if (not path) or (not data):
+        return data
```

Comment:
Would any of these cases ever happen, or is it more of a "just in case"?

Reply:
it's just in case